### PR TITLE
Migrate SnackBar methods to ScaffoldMessenger

### PR DESCRIPTION
**Pull request by Piinks**
cc/ @perclasson and @rami-a contacts for https://github.com/flutter/tests/blob/master/registry/flutter_gallery.test

https://github.com/flutter/flutter/pull/66504 introduced the ScaffoldMessenger widget, which allows for persistent SnackBars across Scaffolds, and easier management of asynchronous SnackBars. The original SnackBar methods in Scaffold are going to be deprecated. Test failures were identified in this repository in https://github.com/flutter/flutter/pull/67947, so this PR is intended to migrate to the new API.

Full Migration Guide:
https://github.com/flutter/website/pull/4527

References: 
https://github.com/flutter/flutter/issues/62921
https://github.com/flutter/flutter/issues/57218

### CODE REVIEW

File: lib/demos/material/dialog_demo.dart [hunk 1]
```diff
@@ -49,8 +50,8 @@ class DialogDemo extends StatelessWidget {
     );
     // The value passed to Navigator.pop() or null.
     if (value != null && value is String) {
-      _scaffoldKey.currentState.hideCurrentSnackBar();
-      _scaffoldKey.currentState.showSnackBar(SnackBar(
+      _scaffoldMessengerKey.currentState.hideCurrentSnackBar();
```
Comment (rami-a) [inline lib/demos/material/dialog_demo.dart:53]: Can this use `ScaffoldMessenger.of(context)` instead?

File: lib/demos/material/dialog_demo.dart [hunk 2]
```diff
@@ -20,6 +20,7 @@ class DialogDemo extends StatelessWidget {
   DialogDemo({Key key, @required this.type}) : super(key: key);
 
   final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
```
Comment (rami-a) [inline lib/demos/material/dialog_demo.dart:22]: Can this key be deleted?

**Review by rami-a — APPROVED** [review]
LGTM, you'll have to run `flutter pub run grinder update-code-segments` to update the demo code snippets and get the CI to pass (see https://github.com/flutter/gallery#generating-highlighted-code-segments for more info).

**Comment by Piinks** [comment]
Thanks for the review @rami-a :)

**Pull request merged**